libs/platform/src/lib/components/form/textarea/textarea.component.ts

```diff
@@ -0,0 +1,354 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ElementRef,
+    Input,
+    Optional,
+    Self,
+    ViewChild,
+    ViewEncapsulation,
+    HostListener
+} from '@angular/core';
+import { NgControl, NgForm } from '@angular/forms';
+import { BaseInput } from '../base.input';
+import { FormFieldControl } from '../form-control';
+
+const VALID_WRAP_TYPES = ['hard', 'soft', 'off'];
+
+export type WrapType = 'hard' | 'soft' | 'off';
+export type FormStates = 'success' | 'error' | 'warning' | 'information' | '';
+
+let uniqueId = 0;
+
+/**
+ * Textarea field implementation to be compliant with our FormGroup/FormField design
+ *
+ */
+@Component({
+    selector: 'fdp-textarea',
+    templateUrl: 'textarea.component.html',
+    styleUrls: ['textarea.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+    providers: [
+        {
+            provide: FormFieldControl,
+            useExisting: TextareaComponent,
+            multi: true
+        }
+    ]
+})
+export class TextareaComponent extends BaseInput {
+    @Input()
+    height: string;
+
+    @Input()
+    growingMaxLines: number; // rows
+
+    @Input()
+    cols: number;
+
+    @Input()
+    wrap: WrapType = 'soft';
+
+    @Input()
+    readonly: boolean;
+
+    @Input()
+    maxLength: number;
+
+    @Input()
+    showExceededText: boolean = false;
+
+    @Input()
+    growing: boolean = false;
+
+    @Input()
+    textExceededMessage: string;
+
+    /**
+     *  The state of the form control - applies css classes.
+     *  Can be `success`, `error`, `warning`, `information` or blank for default.
+     */
+    @Input()
+    stateType: FormStates = 'warning';
+
+    @Input() get value(): any {
+        // if (super.getValue() == null || !super.getValue()) {
+        //     this.value = '';
+        // }
+        return super.getValue();
+    }
+
+    set value(value: any) {
+        if (value) {
+            super.setValue(value);
+        } else {
+            // when custom value not set, we should set/reset counter to maxlength value when it becomes undefined
+            this.exceededCharCount = this.maxLength ? this.maxLength : 0;
+            // reset state by resetting value
+            super.writeValue('');
+        }
+    }
+
+    /** @hidden */
+    @ViewChild('textareaElement')
+    textareaElement: ElementRef;
+
+    /** @hidden */
+    hasTextExceeded: boolean = false;
+
+    /** excess character count */
+    exceededCharCount: number = this.maxLength ? this.maxLength : 0;
+
+    counterExcessOrRemaining = 'remaining';
```

Comment:
I think angular 9 supports i18n inside TS code

Reply:
Since the counter message is dynamic, it can be `10 characters remaining`, or `10 characters over the limit`, or even `1 character remaining` (character vs characters). 
ICU expressions handle this kind of dynamically changing text for i18n by it plural and select expressions. The Select ICU expression expects a variable(in this case `counterExcessOrRemaining`)  which can take some values (in this case `remaining` or `excess`) and we can give the text to be displayed in {} as follows:
```html
{counterExcessOrRemaining, select, excess {over the limit} remaining {remaining}} 
```
the extracted .xlf files will only replace the text in the {} with the respective language.